src/plugins/intel_gpu/src/plugin/transformations/optimize_subsequent_reshapes.hpp

```diff
@@ -0,0 +1,23 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/pass/graph_rewrite.hpp"
+
+namespace ov {
+namespace intel_gpu {
+
+/**
+ * @brief This pass looks for specific patterns of subsequent reshapes that can be
```

Comment:
could you describe a new covered case in this description? 
I believe this is `Reshape [ dynamic dim, n static dims] -> Reshape [dynamic dim, n static dims]`?
can we mention here that  OptimizeSubsequentReshapes can be combined with ReshapeSequenceFusion? mb add a ticket number if any